pkg/config/config.go

```diff
@@ -211,7 +212,15 @@ func (cfg *Config) SetAWSAccountID() error {
 }
 
 // Validate ensures the options are valid
-func (cfg *Config) Validate() error {
+func (cfg *Config) Validate(options ...Option) error {
+	merged := mergeOptions(options)
+	if len(merged.kinds) > 0 {
```

Comment:
I don't really understand the value in the `options` struct? I think it's an early optimisation that isn't adding much - especially since we aren't providing defaults.

Reply:
It's a know paradigm in Go - we could definitely do a follow up to bring default values this way. I agree that it's not bringing that much, but it allows us to add good stuff if needed :)

Reply:
Thinking about https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/config/config.go#L253-L255